File: BaekJoon/BOJ_1920.cpp
#include <cstdio>
#include <algorithm>
using namespace std;

int a[100001];

void bs(int n, int key){
	int start = 0;
	int end = n-1;
	int mid;
	while(start <= end){
		mid = (start + end) / 2;
		
		if(key == a[mid]){
			printf("1\n");
			return ;
		}
		else if(key > a[mid]){
			start = mid+1;
		}
		else{
			end = mid-1;
		}
	}
	printf("0\n");
	return ;
}
 
int main(void){
	int n,m,tmp;
	scanf("%d", &n);
	for(int i=0; i<n; i++){
		scanf("%d", &a[i]);
	}
	sort(a, a+n);
	
	scanf("%d", &m);
	for(int i=0; i<m; i++){
		scanf("%d", &tmp);
		bs(n, tmp);
	}
	
	return 0;
}

File: BaekJoon/길라잡이/BOJ_1920.py
global N_list


def binary_search(N, key):
    global N_list

    start = 0
    end = N - 1

    while start <= end:
        mid = (start + end) // 2

        if key < N_list[mid]:
            end = mid - 1
        elif key > N_list[mid]:
            start = mid + 1
        else:  # x==mid
            return 1
    return 0


def solution():
    global N_list
    N = int(input())
    N_list = list(map(int, input().split()))
    N_list.sort()

    M = int(input())
    M_list = list(map(int, input().split()))

    for x in M_list:
        print(binary_search(N, x))


if __name__ == "__main__":
    solution()
